facebookincubator/facebook-for-woocommerce
# Messenger: pull the latest remote settings

# Summary

Ensure the latest Messenger settings are updated on the site.

### Story: [CH 56455](https://app.clubhouse.io/skyverge/story/56455/facebook-for-woocommerce-settings-not-in-sync-between-fbe-and-messenger-tab)
### Release: #1277 

## Details

In FBE 2 we cannot update the locale, greeting, or color settings from the plugin. This PR removes those settings, and adds the ability to _pull_ the messenger configuration from Facebook. This pull occurs in two ways:

1. Once an hour, sitewide
1. Whenever the Messenger settings page is visited

We _can_ enable or disable the feature from the plugin, so that setting is preserved. Whenever that changes, the FBE configuration is updated.

The messenger JS takes care of applying the color and greeting settings that are configured in FBE for us, so we stop passing those values as attributes when initializing the JS.

The new FBE flow also has a setting for allowing specific domains. If the user accidentally removes the site's domain, the plugin re-adds it whenever pulling the latest settings.

Note: a [later story](https://app.clubhouse.io/skyverge/story/57143/facebook-for-woocommerce-notice-upon-upgrade-about-messenger-customization) will add messaging and any extra handling for upgraders from FBE 1

## UI Changes

### New settings page, messenger enabled
![](https://p-b1Flee.t1.n0.cdn.getcloudapp.com/items/o0uv8eQz/Screen%20Shot%202020-06-17%20at%2011.37.58%20AM.png?v=2612f31787fe33b98f3f0dab2040891f)

### Messenger disabled
![](https://p-b1Flee.t1.n0.cdn.getcloudapp.com/items/KouBrn9r/Screen%20Shot%202020-06-17%20at%201.57.44%20PM.png?v=6dfde02fa0ebfd29dbfede71c0dfa79b)

### Temporary error pulling the latest settings
![](https://p-b1Flee.t1.n0.cdn.getcloudapp.com/items/NQugr4l6/Screen%20Shot%202020-06-17%20at%201.56.47%20PM.png?v=1ce02954b6717451b2ef5f5312544033)

### Temporary error updating the settings remotely
![](https://p-b1Flee.t1.n0.cdn.getcloudapp.com/items/6qu2eNw7/Screen%20Shot%202020-06-17%20at%201.58.29%20PM.png?v=a3fb4e596898b8231cea93d3b64f3285)

## QA

- Disconnect the plugin & clear all DB options

1. Connect
1. Visit the frontend
    - [x] No messenger is displayed
1. From the Connection settings tab, click "Manage Connection"
1. In the FBE UI, click "Add" next to Messenger
1. Configure some custom options, like greeting and color, and complete the wizard
1. Visit the Messenger settings page
    - [x] The Enabled & Locale items are displayed correctly
1. Visit the frontend
    - [x] The messenger is loaded with the correct customizations
1. Visit the Messenger settings again
1. Disable the messenger & save
    - [x] After reload, the setting is still disabled
1. Visit the frontend
    - [x] The messenger is not loaded
1. Enable the messenger and save again
1. Visit the frontend
    - [x] The messenger is loaded

- [x] `run acceptance Admin/Settings` tests pass

## Before merge
